packages/expo/src/hooks/useSSO.ts

```diff
@@ -0,0 +1,104 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { EnterpriseSSOStrategy, OAuthStrategy, SetActive, SignInResource, SignUpResource } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as Linking from 'expo-linking';
+import * as WebBrowser from 'expo-web-browser';
+
+import { errorThrower } from '../utils/errors';
+
+export type UseSSOParams = {
+  strategy: OAuthStrategy | EnterpriseSSOStrategy;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOParams = {
+  identifier?: string;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOFlowReturnType = {
+  /**
+   * Session ID created upon sign-in completion, or null if incomplete.
+   * If incomplete, use signIn or signUp for next steps like MFA.
+   */
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+  authSessionResult?: WebBrowser.WebBrowserAuthSessionResult;
+};
+
+export function useSSO(useSSOParams: UseSSOParams) {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startFlow(startSSOFlowParams: StartSSOParams = {}): Promise<StartSSOFlowReturnType> {
+    if (!isSignInLoaded || !isSignUpLoaded) {
+      return {
+        createdSessionId: null,
+        signIn,
+        signUp,
+        setActive,
+      };
+    }
+
+    let createdSessionId = signIn.createdSessionId;
+
+    const redirectUrl =
+      startSSOFlowParams?.redirectUrl ||
+      useSSOParams.redirectUrl ||
+      AuthSession.makeRedirectUri({
+        path: 'sso-native-callback',
+      });
+
+    await signIn.create({
+      strategy: useSSOParams.strategy,
+      redirectUrl,
+      identifier: startSSOFlowParams.identifier,
+    });
+
+    const { externalVerificationRedirectURL } = signIn.firstFactorVerification;
+    if (!externalVerificationRedirectURL) {
+      return errorThrower.throw('Missing external verification redirect URL for SSO flow');
+    }
+
+    const authSessionResult = await WebBrowser.openAuthSessionAsync(externalVerificationRedirectURL.toString());
```

Comment:
One way to fix deep linking here (more details on this [thread](https://github.com/clerk/javascript/pull/4880/files#r1915799110)) is to call `Linking.openURL` manually on the SSO callback and we'd have to install the `expo-linking' for that. 

I'll proceed with the following:
- Open a issue on the Expo repo linking to a minimum reproduction 
- Open a PR with a patch for both `useSSO` and `useOAuth` using the `Linking` module

Reply:
I've decided to refactor the API to not rely on deep linking on the way back with the main argument of preventing race conditions with `setActive` - here's another comment for reference https://github.com/clerk/javascript/pull/4905#discussion_r1920517176

The consumer should check for `createdSessionId`, or perform the progressive flow, and have total control on navigating on SSO callback